Framework/Kernel/inc/MantidKernel/PropertyWithValue.h

```diff
@@ -637,6 +641,29 @@ template <typename TYPE> class DLLExport PropertyWithValue : public Property {
   PropertyWithValue();
 };
 
+#define PROPERTYWITHVALUE_SAVEPROPERTY(type)                                   \
+  template <>                                                                  \
+  inline void PropertyWithValue<type>::saveProperty(::NeXus::File *file) {     \
+    file->makeGroup(this->name(), "NXlog", 1);                                 \
+    file->writeData("value", m_value);                                         \
+    file->closeGroup();                                                        \
+  }
+
+PROPERTYWITHVALUE_SAVEPROPERTY(float)
+PROPERTYWITHVALUE_SAVEPROPERTY(double)
+PROPERTYWITHVALUE_SAVEPROPERTY(int32_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(uint32_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(int64_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(uint64_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::string)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::vector<double>)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::vector<int32_t>)
+
+template <typename TYPE>
+inline void PropertyWithValue<TYPE>::saveProperty(::NeXus::File *file) {
+  this->saveProperty(file);
+}
```

Comment:
You need to declare the template specialization in the header, but the implementation can be in the source. Also note that you may need DLL macros for the specializations.

The errors look a bit peculiar, not sure what the issue is.